what4/src/What4/Interface.hs

```diff
@@ -700,6 +706,14 @@ class ( IsExpr (SymExpr sym), HashableF (SymExpr sym), HashableF (BoundVar sym)
   --   or for terms that cannot be separated from their annotations.
   getUnannotatedTerm :: sym -> SymExpr sym tp -> Maybe (SymExpr sym tp)
 
+  -- | Make an expression /opaque/, inhibiting rewrites that, in many cases,
+  -- coarsen abstract domain information. See #248 for discussion and #249 for
+  -- progress towards obviating opaque expressions.
+  opacify :: sym -> SymExpr sym tp -> IO (SymExpr sym tp)
+
+  -- | Inverse operation of 'opacify'.
+  clarify :: sym -> SymExpr sym tp -> SymExpr sym tp
```

Comment:
It would be helpful to specify what happens if you call `clarify` on a `SymExpr` that wasn't returned by a call to `opacify`.